src/us/kbase/workspace/modules/KafkaNotifierFactory.java

```diff
@@ -0,0 +1,297 @@
+package us.kbase.workspace.modules;
+
+import static java.util.Objects.requireNonNull;
+import static us.kbase.workspace.database.Util.checkString;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Optional;
+
+import us.kbase.workspace.database.ObjectInformation;
+import us.kbase.workspace.database.Permission;
+import us.kbase.workspace.database.WorkspaceUser;
+import us.kbase.workspace.listener.ListenerInitializationException;
+import us.kbase.workspace.listener.WorkspaceEventListener;
+import us.kbase.workspace.listener.WorkspaceEventListenerFactory;
+
+// TODO NOW add example string once all fields are done
+/** A workspace listener that sends workspace events to Kafka as JSON strings.
+ * 
+ * The listener requires two configuration arguments:
+ * topic - the topic to which the listener will submit events. The listener requires the topic
+ * name to consist of ASCII alphanumeric values and the hyphen to avoid Kafka issues around
+ * ambiguity between period and underscore values. 
+ * bootstrap.servers - the Kafka bootstrap servers.
+ * 
+ * The listener is configured to require a full write to all the replicates before a call
+ * to Kafka returns, and if a write fails, an exception is thrown in the thread that called
+ * the listener.
+ * 
+ * @author gaprice@lbl.gov
+ *
+ */
+public class KafkaNotifierFactory implements WorkspaceEventListenerFactory {
+
+		// TODO DOCS
```

Comment:
Some docs about the JSON object that you are pushing into kafka would be very helpful - even just for a reviewer of your PR.

Reply:
I'm not going to write any docs until I have the JSON object figured out for all the required events, but if you have any specific questions I can answer them.

The current format looks like this:
```
{"wsid":1,"ver":1,"evtype":"NEW_VERSION","objid":16,"time":1547754188639,"objtype":"Empty.AType-1.0","user":"gaprice"}
```

Reply:
It doesn't need to be a formal document. Just comments in the code would be good. I think that Shane, Bill and yourself are all working on kafka messages, putting a comment in the code that just shows the current state of the JSON object would make it easier for all of you to peek over each other's shoulders and see where things are headed.

Reply:
This is going to be done today or tomorrow and I'll document it at that point, and it's documented above. It's WIP mang

Reply:
Alternatively, here's the documentation in the code: https://github.com/kbase/workspace_deluxe/pull/225/files#diff-2d3262cb67db8004f2545742863b5e42R285